dask_kubernetes/operator/controller/controller.py

```diff
@@ -381,6 +382,75 @@ async def retire_workers(
         return [w["metadata"]["name"] for w in workers.items[:-n_workers]]
 
 
+async def check_scheduler_idle(scheduler_service_name, namespace, logger):
+    # Try getting idle time via HTTP API
+    dashboard_address = await get_scheduler_address(
+        scheduler_service_name,
+        namespace,
+        port_name="http-dashboard",
+        allow_external=False,
+    )
+    async with aiohttp.ClientSession() as session:
+        url = f"{dashboard_address}/api/v1/check_idle"
```

Comment:
The current `check_idle` impl on the scheduler looks like it will close itself via a background asyncio task if it passes the idle timeout from the regular dask config.

If a user happened to have `distributed.scheduler.idle-timeout` set, this might be confusing if their scheduler is exits due to idleness but the dask cluster resource isn't able to pick up on why it closed and then trigger the k8s dask cluster cleanup in response.

Another option could be to use the --death-timeout concept that workers have and auto-delete the dask cluster if the scheduler dies and stays dead for some period of time. This could help cleanup dask cluster k8s resources in cases the scheduler is unrecoverable and dask cluster operator won't be able to ever make a call to check idleness.

Reply:
I hear what you're saying. With this implementation, if the scheduler dies or becomes unresponsive we will never clean it up.

Today if the scheduler exits/dies it doesn't get recreated and the `DaskCluster` goes into a broken state. In #606 we discuss recreating the scheduler (probably using a `Deployment` or `ReplicaSet`). This is the opposite of what I understand you are proposing here which is cleaning up the whole `DaskCluster` if the scheduler is lost.

My view is that if a `DaskCluster` resource exists the controller should be working to get everything into a working state. We can achieve this via liveness/readiness probes and recreating missing `Pods`. If the scheduler or workers die it should be trying to reconcile and repair the cluster. We have some work to do to get to this place though.

I think actively reconciling the cluster and then asking the scheduler if it wants to time out is the right approach for this PR. But perhaps we also want some other timeout where if the controller can't get the cluster into a healthy state for a certain amount of time then it gives up and cleans it up.

Reply:
These are great points. The PR as is looks like it would alleviate nearly all of the manual work we're currently facing with regards to cleanup.

Very excited for it to land!